dagmc_stats.py

```diff
@@ -300,3 +300,184 @@ def get_coarseness(my_core, meshset, entity_ranges, geom_dim):
         coarseness.append(len(surf_area)/sum(surf_area))
 
     return coarseness
+
+
+def get_beta_angles(my_core, pair_tris, common_vert, beta_angles):
+    """
+    Gets the beta angles of given triangle pairs
+
+    inputs
+    ------
+    my_core : a MOAB Core instance
+    pair_tris : triangle pair
+    common_vert : common vertices in the triangle pair
+    beta_angles: list that stores the beta angles of given triangle pair
+    """
+    vert1 = common_vert[0]
+    vert2 = common_vert[1]
+    side_lengths = []
+    
+    for m in range(2):
+        adj_verts = my_core.get_adjacencies(pair_tris[m], 0, op_type=1)
+        for n in range(3):
+            if adj_verts[n] != vert1 and adj_verts[n] != vert2:
+                other_vert = adj_verts[n]        
+        side_lengths.append(np.linalg.norm(my_core.get_coords(vert1)-my_core.get_coords(vert2)))
+        side_lengths.append(np.linalg.norm(my_core.get_coords(vert1)-my_core.get_coords(other_vert)))
+        side_lengths.append(np.linalg.norm(my_core.get_coords(vert2)-my_core.get_coords(other_vert)))
+        beta_angles.append(np.arccos((side_lengths[1] * side_lengths[1]
+                            + side_lengths[2] * side_lengths[2]
+                            - side_lengths[0] * side_lengths[0])
+                            /(2.0 * side_lengths[1] * side_lengths[2])))
+    
+
+def get_angles(my_core, tri, vert = None):
+    """
+    Gets the angles of the given triangle
+
+    inputs
+    ------
+    my_core : a MOAB Core instance
+    tri: triangle
+    vert: the vertex which generates the triangle by get_adj
+
+    outputs
+    -------
+    angles : (list) the angles in the tri
+    """
+    
+    angles = []
+    side_lengths = []
+    coord_list = []
+    index = 3
+    if vert != None:
+        verts = my_core.get_adjacencies(tri, 0, op_type=1)
+        for i in range(len(verts)):
+            if verts[i] == vert:
+                index = i
+        for vert in verts:
+            coords = my_core.get_coords(vert)
+            coord_list.append(coords)
+        side_lengths.append(np.linalg.norm(coord_list[index]-coord_list[index-1]))
+        side_lengths.append(np.linalg.norm(coord_list[index]-coord_list[index-2]))
+        side_lengths.append(np.linalg.norm(coord_list[index-1]-coord_list[index-2]))
+    else:
+        side_lengths = get_tri_side_length(my_core, tri)
+    for side in range(3):
+        angles.append(np.arccos((side_lengths[side] * side_lengths[side]
+                            + side_lengths[side - 2] * side_lengths[side - 2]
+                            - side_lengths[side - 1] * side_lengths[side - 1])
+                            /(2.0 * side_lengths[side] * side_lengths[side - 2])))
```

Comment:
Almost certainly better here to do something like:
```
side_lengths_sq = np.dot(side_lengths,side_lengths)
side_length_prod = np.prod(side_lengths)
for this_side_length in side_lengths:
    cos_angle = (side_lengths_sq - 2.0*this_side_length*this_side_length)/(2.0 * side_length_prod/this_side_length)
    angles.append(np.arccos(cos_angle))
```

Reply:
In order to know which angle is the alpha or beta angle, I need to preserve the order of the angles. This seems to be calculating the cos values in other order and therefore does not work.